source/read/change-streams.txt

```diff
@@ -0,0 +1,284 @@
+   :depth: 2
+   :class: singlecol
+
+.. facet::
+   :name: genre
+   :values: reference
+
+.. meta::
+   :keywords: watch, code example
+
+Overview
+--------
+
+In this guide, you can learn how to use the {+driver-short+} to monitor a  **change stream**,
+allowing you to view real-time changes to your database. A change stream is a {+mdb-server+} feature that
+publishes data changes on a collection, database, or deployment. Your application can
+subscribe to a change stream and use events to perform other actions.
+
+Sample Data
+~~~~~~~~~~~
+
+The examples in this guide use the ``restaurants`` collection in the ``sample_restaurants``
+database from the :atlas:`Atlas sample datasets </sample-data>`. To learn how to create a
+free MongoDB Atlas cluster and load the sample datasets, see the
+:atlas:`Get Started with Atlas </getting-started>` guide.
+
+The following {+language+} data class models the documents in this collection:
+
+.. literalinclude:: /includes/read/change-streams.kt
+   :start-after: start-data-class
+   :end-before: end-data-class
+   :language: kotlin
+   :copyable:
+
+Open a Change Stream
+--------------------
+
+To open a change stream, call the ``watch()`` method. The instance on which you
+call the ``watch()`` method on determines the scope of events that the change
+stream listens for. You can call the ``watch()`` method on instances of the following
+classes:
+
+- ``MongoClient``: To monitor all changes in the MongoDB deployment
+- ``MongoDatabase``: To monitor changes in all collections in the database
+- ``MongoCollection``: To monitor changes in the collection
+
+Open a Change Stream Example
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+The following example opens a change stream on the ``restaurants`` collection
+and prints changes as they occur:
+
+.. literalinclude:: /includes/read/change-streams.kt
+   :start-after: start-open-change-stream
+   :end-before: end-open-change-stream
+   :language: kotlin
+   :copyable:
+   :dedent:
+
+To begin watching for changes, run the application. Then, in a separate
+application or shell, perform a write operation on the ``restaurants`` collection. The
+following example updates a document in which the value of the ``name`` is ``"Blarney Castle"``:
+
+.. _kotlin-sync-change-stream-update:
+
+.. literalinclude:: /includes/read/change-streams.kt
+   :start-after: start-update-for-change-stream
+   :end-before: end-update-for-change-stream
+   :language: kotlin
+   :copyable:
+   :dedent:
+
+When you update the collection, the change stream application prints the change
+as it occurs. The printed change event resembles the
+following:
+
+.. code-block:: json
+   :emphasize-lines: 14-20
+
+   {
+     "cursor": {
+       "nextBatch": [
+         {
+           "_id": { ... },
+           "operationType": "update",
+           "clusterTime": { ... },
+           ...
+           "ns": {
+             "db": "sample_restaurants",
+             "coll": "restaurants"
+           },
+           ...
```

Comment:
I'm not clear on why only the updateDescription is being highlighted.  It would make sense if this section was discussing the updateDescription in particular, but it is not.  Moreover, below we show an example of filtering on the operationType, and that is not highlighted.

I suggest that the highlighting be removed, and also remove this from the top:

```
   {
     "cursor": {
       "nextBatch": [
```

and this from the bottom:

```
       ],
       ...
     }
   }
```

The reason is that the cursor/nextBatch wrapper is part of the cursor wire protocol and is never actually visible to the application.

Reply:
Will apply these suggestions to the next revision!